wsd/COOLWSD.cpp

```diff
@@ -511,8 +511,11 @@ static bool prespawnChildren()
 
 #endif
 
-static size_t addNewChild(const std::shared_ptr<ChildProcess>& child)
```

Comment:
This is all fine of course, though I've no problem with const & generally.

Reply:
The reason for the change isn't `const&` it's when the `ChildProcess` instance is copied. Specifically, with the copy taken at the call-site, the copying is done outside the lock. With the `const&`, copying is done while the lock is taken, extending the duration of the lock and making contention more likely. I also reduced logging while the lock is taken for the same reason.